dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/data/DefaultQueryPlanner.java

```diff
@@ -152,12 +152,34 @@ public DataQueryParams withTableNameAndPartitions( DataQueryParams params, Query
             PartitionUtils.getPartitions( params.getStartDate(), params.getEndDate() ) :
             PartitionUtils.getPartitions( params.getAllPeriods() );
 
+        if ( params.getCurrentUser() != null )
```

Comment:
Just wondering, why do we check for current user here?

Reply:
yes, that is because the tests would fail otherwise. 

What we do during test is simply check number of partitions. However, partitionManager.filterNonExistingPartitions removes those partitions that do not exist physically ... which we don't have during testing.